src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,29 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector, loadingText }) {
+    super({ popupSelector });
+    this._loadingText = loadingText;
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._confirmDeleteButton =
+      this._popupElement.querySelector("#modal-delete-btn");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._confirmDeleteButton.textContent = this._loadingText;
+    } else {
+      this._confirmDeleteButton.textContent = this._submitButtonText;
+    }
+  }
+
+  open() {
+    this._popupForm.addEventListener("submit", this._handleFormSubmit);
```

Comment:
It's also needed to remove the listener when closing the  popup

Reply:
I don't understand what you are saying here? What's also needed?

Reply:
Are you saying you want the below added to popupWithConfirmation?

close() {
    this._popupForm.removeEventListener("submit", this._handleSubmit);
    this._popupForm.reset();
    super.close();
  }

Reply:
Done